rust/noosphere-gateway/src/single_tenant/job_client.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{
+    extractors::GatewayScope,
+    jobs::{
+        worker_queue::{WorkerQueue, WorkerQueueBuilder},
+        GatewayJob, GatewayJobContext, GatewayJobProcessor, JobClient,
+    },
+    SingleTenantContextResolver,
+};
+use anyhow::Result;
+use noosphere_core::context::HasMutableSphereContext;
+use noosphere_ipfs::KuboClient;
+use noosphere_ns::server::HttpClient as NameSystemHttpClient;
+use noosphere_storage::Storage;
+use std::{marker::PhantomData, sync::Arc, time::Duration};
+use tokio::task::JoinHandle;
+use url::Url;
+
+#[cfg(doc)]
+use crate::single_tenant::SingleTenantGatewayManager;
+
+/// The number of worker threads to process jobs.
+const WORKER_COUNT: usize = 5;
+/// How many times by default jobs should retry upon failure.
+/// @TODO Consider allowing individual jobs to hint if they
+/// should be retried, versus failures that can be attempted
+/// on next periodic cycle.
+const JOB_RETRIES: usize = 1;
+/// How many seconds before a job is considered broken,
+/// and potentially restarted.
+const TIMEOUT_SECONDS: u64 = 180;
```

Comment:
Syndicating large histories could be longer than this timeout, and notably, name records' publishing timeframe is currently 2 minutes.